pkg/operator/controller/sync-http-error-code-configmap/controller.go

```diff
@@ -0,0 +1,239 @@
+		GenericFunc: func(e event.GenericEvent) bool { return reconciler.configmapIsInUse(e.Meta) },
+	}); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+type reconciler struct {
+	client                   client.Client
+	recorder                 record.EventRecorder
+	operatorNamespace        string
+	cache                    cache.Cache
+	operandNamespace         string
+	sourceConfigMapNamespace string
+}
+
+// configmapToIngressController maps a secret to a slice of reconcile requests,
+// one request per ingresscontroller that references the configmap.
+
+func (r *reconciler) configmapToIngressController(o handler.MapObject) []reconcile.Request {
+	requests := []reconcile.Request{}
+	controllers, err := r.ingressControllersWithConfigMap(o.Meta.GetName())
+	if err != nil {
+		log.Error(err, "failed to list ingresscontrollers for configmap", "related", o.Meta.GetSelfLink())
+		return requests
+	}
+	for _, ic := range controllers {
+		log.Info("queueing ingresscontroller", "name", ic.Name, "related", o.Meta.GetSelfLink())
+		request := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: ic.Namespace,
+				Name:      ic.Name,
+			},
+		}
+		requests = append(requests, request)
+	}
+	return requests
+}
+
+// ingressControllersWithConfigMap returns the ingresscontrollers that reference
+// the given configmap.
+func (r *reconciler) ingressControllersWithConfigMap(configmapName string) ([]operatorv1.IngressController, error) {
+	controllers := &operatorv1.IngressControllerList{}
+	if err := r.cache.List(context.Background(), controllers); err != nil {
+		return nil, err
+	}
+	return controllers.Items, nil
+}
+
+// configmapIsInUse returns true if the given secret is referenced by some
+// ingresscontroller.
+func (r *reconciler) configmapIsInUse(meta metav1.Object) bool {
+	controllers, err := r.ingressControllersWithConfigMap(meta.GetName())
+	if err != nil {
+		log.Error(err, "failed to list ingresscontrollers for configmap", "related", meta.GetSelfLink())
+		return false
+	}
+	return len(controllers) > 0
+}
+
+// hasConfigMap returns true if the effective  httpErrorCodePage configmap for the
+// given ingresscontroller exists, false otherwise.
+func (r *reconciler) hasConfigMap(meta metav1.Object, o runtime.Object) bool {
+	ic := o.(*operatorv1.IngressController)
+	configMapName := controller.HttpErrorCodePageConfigMapName(ic, r.operandNamespace)
+	configMap := &corev1.ConfigMap{}
+	if err := r.client.Get(context.Background(), configMapName, configMap); err != nil {
+		if errors.IsNotFound(err) {
+			return false
+		}
+		log.Error(err, "failed to look up configmap for ingresscontroller", "name", configMapName, "related", meta.GetSelfLink())
+	}
+	return true
+}
+
+// configMapChanged returns true if the name of config
+//for the given ingresscontroller has changed, false
+// otherwise.
+//TODO check the data between old and new configmaps
+func (r *reconciler) configMapChanged(old, new runtime.Object) bool {
```

Comment:
Duplicate function?

Reply:
Actually I need them for each for both controllers

Reply:
HttpErrorCodePageConfigMapName  is actually used by currentHttpErrorCodeConfigMap  . they both return different number of arguments.